dataland-frontend/tests/e2e/specs/search-company-framework-data/ViewMyDatasets.ts

```diff
@@ -0,0 +1,155 @@
+          cy.visitAndCheckAppMount("/companies");
+          validateTabBar(0);
+          cy.wait(3000);
+          const companiesInterceptAlias = "companies";
+          cy.intercept("**/api/companies*onlyCurrentUserAsUploader*").as(companiesInterceptAlias);
+          cy.get(getTabSelector(1)).click();
+          cy.wait(`@${companiesInterceptAlias}`, { timeout: Cypress.env("long_timeout_in_ms") as number });
+          cy.url().should("contain", "/datasets");
+          validateTabBar(1);
+
+          cy.get(newDatasetButtonSelector).should("exist");
+
+          const expectedHeaders = ["COMPANY", "DATA FRAMEWORK", "SUBMISSION DATE"];
+          expectedHeaders.forEach((value) => {
+            cy.get(`table th:contains(${value})`).should("exist");
+          });
+          const unexpectedHeaders = ["YEAR"];
+          unexpectedHeaders.forEach((value) => {
+            cy.get(`table th:contains(${value})`).should("not.exist");
+          });
+          cy.get("th").each((element) => {
+            if (!expectedHeaders.includes(element.text())) {
+              expect(element.html()).to.contain("<input");
+            }
+          });
+        });
+
+        it("Check if new dataset button leads to choose company page and if the back button works as expected", () => {
+          cy.ensureLoggedIn(uploader_name, uploader_pw);
+          cy.visit("/datasets");
+          validateTabBar(1);
+          cy.get(newDatasetButtonSelector).click();
+          cy.url().should("contain", "/companies/choose");
+          cy.get("[title='back_button']").click();
+          cy.url().should("contain", "/datasets");
+        });
+
+        it("Check that a user who has no dataset associated with him has no table displayed", () => {
+          cy.ensureLoggedIn(admin_name, admin_pw);
+          cy.visit("/datasets");
+          cy.get(newDatasetButtonSelector).should("exist");
+          cy.get("table").parent().parent().parent().should("have.class", "hidden");
+        });
+
+        it("Check that a user who has no upload permission has no new dataset button displayed", () => {
+          cy.ensureLoggedIn(admin_name, admin_pw);
+          cy.visit("/datasets");
+          cy.get(newDatasetButtonSelector).should("not.exist");
+        });
+
+        let newCompanyName = "";
+        it("Check if search filter works as expected", () => {
+          cy.ensureLoggedIn(uploader_name, uploader_pw);
+          cy.visit("/datasets");
+          newCompanyName = `Overview Search ${faker.company.bsNoun()}`;
+          cy.get(searchBarSelector).type(newCompanyName);
+          cy.get("tbody td").then((elements) => {
+            expect(elements.length).to.equal(1);
+          });
+          uploadCompanyWithLksgDataset(newCompanyName);
+          cy.reload();
+          cy.visit("/datasets");
+          cy.get(searchBarSelector).type(newCompanyName);
+          cy.get("tbody td").then((elements) => {
+            expect(elements.length).to.equal(4);
```

Comment:
Can someone explain me what is happening here?

Reply:
The test checks first if specifically earching for a non-existing company yields an empty table. The check for 1 element of type "tbody td" is done because if the table is empty there is still one row displayed stretched over the whole length of the table. Then it uploads this company and expects to see one entry in the table. There shall also be 4 columns in this row and therefore also in the body (which only has this single row).

Reply:
Ok, Thank you